### PR TITLE
Add tests targeting core functionality of `/data_objects/study/{study_id}` endpoint

### LINKED ISSUE #725: Core functionality of `/data_objects/study/{study_id}` endpoint is not being tested
The only test that targets the `/data_objects/study/{study_id}` endpoint, targets a corner case where the specified study does not exist.

https://github.com/microbiomedata/nmdc-runtime/blob/bcdf8f390974786f879806123b28f926881b056c/tests/test_api/test_endpoints.py#L349-L364

There are no tests that target the core functionality (a.k.a. "happy path") of the endpoint.

https://github.com/microbiomedata/nmdc-runtime/blob/bcdf8f390974786f879806123b28f926881b056c/tests/test_api/test_endpoints.py#L36-L38

Previously, there was a test that targeted the core functionality; but it was removed because it relied on a static database dump (which conformed to a specific schema) residing on the NERSC filesystem.

Related (bug report involving this endpoint): #723

### PR DESCRIPTION
**Pull request by eecavanna**
> [!NOTE]
> This branch is under construction.

In this branch, I implemented some pytest tests that exercise the core functionality of the `/data_objects/study/{study_id}` endpoint.

### Details

The tests do not currently exercise _all_ of the core functionality of that endpoint. I plan to hand this branch off to someone else to fill in the gaps (by following the "example" tests I have added here).

### Related issue(s)

Eventually, this PR will fix #725.

### Related subsystem(s)

- [x] Runtime API (except the Minter)
- [ ] Minter
- [ ] Dagster
- [ ] Project documentation (in the `docs` directory)
- [ ] Translators (metadata ingest pipelines)
- [ ] MongoDB migrations
- [ ] Other

### Testing

- [x] I tested these changes (explain below)
- [ ] I did not test these changes

I tested these changes by running `$ make test-run` on this branch, where the `test-run` Make target has been redefined locally as:

```make
test-run:
	docker compose --file docker-compose.test.yml run test tests/test_api/test_endpoints.py
```

### Documentation

- [x] I **have not checked** for relevant documentation yet (e.g. in the `docs` directory)
- [ ] I have **updated** all relevant documentation so it will remain accurate
- [ ] Other (explain below)

### Maintainability

- [x] Every Python function I defined includes a docstring _(test functions are exempt from this)_
- [x] Every Python function parameter I introduced includes a type hint (e.g. `study_id: str`)
- [x] All "to do" or "fix me" Python comments I added begin with either `# TODO` or `# FIXME`
- [x] I used `black` to format all the Python files I created/modified
- [x] The PR title is in the imperative mood (e.g. "Do X") and not the declarative mood (e.g. "Does X" or "Did X")

### CODE REVIEW

**Comment by eecavanna** [comment]
Hi @sujaypatil96, I created this branch on which I've written some tests that target some of the core functionality of the `/data_objects/study/{study_id}` endpoint. I'll bring this up during this week's referential integrity squad meeting.

File: tests/test_api/test_endpoints.py [hunk 1]
```diff
@@ -29,19 +29,23 @@
...
 
 
-def ensure_schema_collections_and_alldocs():
-    # Return if `alldocs` collection has already been materialized.
+def ensure_schema_collections_and_alldocs(force_refresh_of_alldocs: bool = False):
+    r"""
+    This function can be used to ensure things (?) about schema-described collections and the "alldocs" collection.
+
+    :param bool force_refresh_of_alldocs: Whether you want to force a refresh of the "alldocs" collection,
+                                          regardless of whether it is empty of not. By default, this function
```
Comment (eecavanna) [inline tests/test_api/test_endpoints.py:37]:
```suggestion
                                          regardless of whether it is empty or not. By default, this function
```

**Comment by sujaypatil96** [comment]
Mermaid diagram describing instance relationships in the example/scenario used in `test_find_data_objects_for_study_having_one()`:

```mermaid
graph LR
    nmdc:bsm-11-6zd5nb38 -- "associated_studies" --> nmdc:sty-11-r2h77870
    nmdc:bsm-11-6zd5nb38 -- "has_input" --> nmdc:omprc-11-nmtj1g51
    nmdc:omprc-11-nmtj1g51 -- "was_generated_by" --> nmdc:dobj-11-cpv4y420

```

**Comment by dwinston** [comment]
@sujaypatil96 I think our "setup" is incomplete. It seems like there needs to be a `workflow_execution_set` entity for `GET /data_objects/study/{study_id}` to find properly.

**Comment by sujaypatil96** [comment]
@dwinston yes you're right! This is what the `workflow_execution_set` record would look like:

```
workflow_execution_dict =  {"id" : "nmdc:wfmsa-11-fqq66x60.1",
    "was_informed_by" : "nmdc:omprc-11-nmtj1g51",
    "has_input" : [
        "nmdc:bsm-11-6zd5nb38"
    ],
    "has_output" : [
        "nmdc:dobj-11-cpv4y420"
    ],
    "type" : "nmdc:MetagenomeSequencing"
}
```

**Comment by dwinston** [comment]
this is ready for review and merge.

**Review by sujaypatil96 — APPROVED** [review]
There's a test case for the `/data_objects/study/{study_id}` endpoint in the test suite now. There's a [mermaid diagram](https://github.com/microbiomedata/nmdc-runtime/pull/849#issuecomment-2555602038) that captures the scenario we are simulating.

We're basically looking for an expected `DataObject` id in the response JSON for the `/data_objects/study/{study_id}` for a given study id and that works.
